src/main/java/core/basesyntax/storage/impl/Database.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.storage.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Database<T> {
+    private final Map<T, Integer> productBalance;
+
+    public Database() {
+        productBalance = new HashMap<>();
+    }
+
+    public Map<T, Integer> getBalance() {
+        return Map.copyOf(productBalance);
+    }
+
+    public int getAmount(T objectT) {
+        return productBalance.get(objectT);
+    }
+
+    public void updateBalance(T objectT, int amount) {
+        productBalance.put(objectT, amount);
+    }
+}
```

Comment:
this abstraction is not appropriate for this task
we are creating fruit shop that works with fruits, no need to overcomplicate things and attach the fifth wheel to the cart :)
```suggestion
public class Database {
    private static final Map<Fruit, Integer> database = new HashMap<>();
}
```